build.cmd

```diff
@@ -390,13 +390,24 @@ if %__RestoreOptData% EQU 1 (
 )
 
 REM Parse the optdata package versions out of msbuild so that we can pass them on to CMake
-for /f "tokens=*" %%s in ('call "%__ProjectDir%\dotnet.cmd" msbuild "%OptDataProjectFilePath%" /t:DumpPgoDataPackageVersion /nologo') do (
-    set __PgoOptDataVersion=%%s
+call "%__ProjectDir%\dotnet.cmd" msbuild "%OptDataProjectFilePath%" /t:DumpPgoDataPackageVersion /nologo %__CommonMSBuildArgs%
+
+if not exist "%__IntermediatesDir%\optdataversion.txt" (
```

Comment:
Also check the errorlevel from the build call? The command could fail, but the file could still exist from a previous build.

Reply:
Addressed